doc/guides/timers-in-node.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Timers in Node.js
+layout: docs.hbs
+---
+
+# Timers in Node.js and beyond
+
+The Timers module in Node.js contains functions that execute code after a set
+period of time. Timers do not need to be imported via `require()`, since
+all the methods are available globally to emulate the browser JavaScript API.
+To fully understand when timer functions will be executed, it's a good idea to
+read up on the the Node.js
+[Event Loop](https://nodesource.com/blog/understanding-the-nodejs-event-loop/).
+
+## Controlling the Time Continuum with Node.js
+
+The Node.js API provides several ways to of scheduling code to execute at
+some point after the present moment. The functions below may seem familiar,
+since they are available in most browsers, but Node.js actually provides 
+it's own implementation of these methods. Timers integrate very closely 
+with the system, and despite the fact that the API mirrors the browser 
+API, there are some differences in implementation.
+
+### "When I say so" Execution ~ *`setTimeout()`*
+
+`setTimeout()` can be used to schedule code execution after a designated
+amount of milliseconds. This function is also a part of the browser 
+JavaScript API and can be used in the exact same manner.
+
+`setTimeout()` accepts the code to execute as its first argument and the
+millisecond delay defined as a number literal as the second argument. Additional
+arguments may also be included and these will be passed on to the function. Here
+is an example of that:
+
+```js
+function myFunc (arg) {
+  console.log('arg was => ' + arg);
+}
+
+setTimeout(myFunc, 1500, 'funky');
+```
+
+The above function `myFunc()` will execute as close to 1500
+milliseconds (or 1.5 seconds) as possible due to the call of `setTimeout()`.
+
+The timeout interval that is set cannot be relied upon to execute after
+that *exact* number of milliseconds. This is because other executing code that
+blocks or holds onto the event loop will push the execution of the timeout
+back. The *only* guarantee is that the timeout will not execute *sooner* than
+the declared timeout interval.
+
+`setTimeout()` returns a `Timeout` object that can be used to reference the
+timeout that was set. This returned object can be used to cancel the timeout (
+see `clearTimeout()` below) as well as change the execution behavior (see
+`unref()` below).
+
+### "Right after this" Execution ~ *`setImmediate()`*
+
+`setImmediate()` will execute code at the beginning of the next
+event loop cycle. This code will execute *before* any timers or IO operations.
```

Comment:
More like at the end of the current one, see https://github.com/nodejs/node/blob/master/doc/topics/the-event-loop-timers-and-nexttick.md#setimmediate-vs-settimeout